server/api/jellyfin.ts

```diff
@@ -0,0 +1,264 @@
+  private userId?: string;
+  private jellyfinHost: string;
+  private axios: AxiosInstance;
+
+  constructor(jellyfinHost: string, authToken?: string, userId?: string) {
+    this.jellyfinHost = jellyfinHost;
+    this.authToken = authToken;
+    this.userId = userId;
+
+    let authHeaderVal = '';
+    if (this.authToken) {
+      authHeaderVal = `MediaBrowser Client="Overseerr", Device="Axios", DeviceId="TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6ODUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC84NS4wfDE2MTI5MjcyMDM5NzM1", Version="10.8.0", Token="${authToken}"`;
+    } else {
+      authHeaderVal =
+        'MediaBrowser Client="Overseerr", Device="Axios", DeviceId="TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6ODUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC84NS4wfDE2MTI5MjcyMDM5NzM1", Version="10.8.0"';
+    }
+
+    this.axios = axios.create({
+      baseURL: this.jellyfinHost,
+      headers: {
+        'X-Emby-Authorization': authHeaderVal,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    });
+  }
+
+  public async login(
+    Username?: string,
+    Password?: string
+  ): Promise<JellyfinLoginResponse> {
+    try {
+      const account = await this.axios.post<JellyfinLoginResponse>(
+        '/Users/AuthenticateByName',
+        {
+          Username: Username,
+          Pw: Password,
+        }
+      );
+      return account.data;
+    } catch (e) {
+      throw new Error('Unauthorized');
+    }
+  }
+
+  public async getServerName(): Promise<string> {
+    try {
+      const account = await this.axios.get<JellyfinUserResponse>(
+        `/System/Info/Public'}`
+      );
+      return account.data.ServerName;
+    } catch (e) {
+      logger.error(
+        `Something went wrong while getting the server name from the Jellyfin server: ${e.message}`,
+        { label: 'Jellyfin API' }
+      );
+      throw new Error('girl idk');
+    }
+  }
+
+  public async getUser(): Promise<JellyfinUserResponse> {
+    try {
+      const account = await this.axios.get<JellyfinUserResponse>(
+        `/Users/${this.userId ?? 'Me'}`
+      );
+      return account.data;
+    } catch (e) {
+      logger.error(
+        `Something went wrong while getting the account from the Jellyfin server: ${e.message}`,
+        { label: 'Jellyfin API' }
+      );
+      throw new Error('Invalid auth token');
+    }
+  }
+
+  public async getLibraries(): Promise<JellyfinLibrary[]> {
+    try {
+      const account = await this.axios.get<any>('/Library/MediaFolders');
+
+      const response: JellyfinLibrary[] = [];
+
+      account.data.Items.forEach((Item: any) => {
+        const library: JellyfinLibrary = {
+          key: Item.Id,
+          title: Item.Name,
+          type: Item.CollectionType == 'movies' ? 'movie' : 'show',
+          agent: 'jellyfin',
+        };
+
+        if (Item.Type == 'CollectionFolder') {
+          response.push(library);
+        }
+      });
```

Comment:
```suggestion
      const response: JellyfinLibrary[] = account.data.Items.filter(
        (Item: any) => {
          return (
            Item.Type == 'CollectionFolder' &&
            (Item.CollectionType == 'tvshows' ||
              Item.CollectionType == 'movies')
          );
        }
      ).map((Item: any) => {
        return <JellyfinLibrary>{
          key: Item.Id,
          title: Item.Name,
          type: Item.CollectionType == 'movies' ? 'movie' : 'show',
          agent: 'jellyfin',
        };
      });
```
We also need to make sure that the library is not mixed content (like the Collection library would be); only `Item.CollectionType` of `tvshows` and `movies`.